data_source_obmcs_core_instance_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	baremetal "github.com/oracle/bmcs-go-sdk"
+
+	"github.com/stretchr/testify/suite"
+	"fmt"
+	"time"
+)
+
+type DatasourceCoreInstanceTestSuite struct {
+	suite.Suite
+	Client 			*baremetal.Client
+	Config			string
+	Provider		terraform.ResourceProvider
+	Providers 		map[string]terraform.ResourceProvider
+	ResourceName	string
+}
+
+
+var instance_display_name string
+var metadata_ssh_key_entry = "ssh_authorized_keys"
+
+func timestampNameSuffix() string {
+	var t = time.Now()
+	return fmt.Sprintf("%d%02d%02d%02d%02d%02d", t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second())
+}
```

Comment:
Recommend calling this `getTimestamp` and putting it in a new file called `test_helpers.go`